cmd/common/utils/version-checker.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/inspektor-gadget/inspektor-gadget/internal/version"
+)
+
+func CheckServerVersionSkew(serverVersion string) error {
+	clientSemver := version.Version()
+
+	// Do not print any warning if this is a prerelease to avoid annoying developers
+	if len(clientSemver.Pre) > 0 {
+		return nil
+	}
+
+	serverSemver, err := semver.ParseTolerant(serverVersion)
+	if err != nil {
+		log.Debugf("parsing server version: %s", err)
+		return nil
+	}
```

Comment:
Ideally we should, but then it'll cause the "Check that README is up-to-date" CI job to fail because the error will be printed. It happens because in the CI the tags aren't available, so `VERSION` can be parsed to a semver as described in https://github.com/inspektor-gadget/inspektor-gadget/pull/2446#issuecomment-1966828604. 

There are few options to fix it:
- fetch the tags on the CI 
- implement a different way to get the version without depending on git

But all of them seem to complicated to me, so ignoring this error should be fine for this "non-essential" feature. WDYT?

Reply:
> Lines do not match: "time="2024-04-16T12:50:20Z" level=warning msg="parsing semver: Version string empty"" != "Collection of gadgets for Kubernetes developers" at tools/check-readme.pl line 80.

From https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/8705591124/job/23877036562